### DISCUSSION TITLE
Automatically update the LSP config when `rome.json` changes

### DISCUSSION THREAD

**Discussion by lgarron** [rome/tools#3216] [Suggestions]
### Environment information

```bash
macOS 12.5.1
VSCode v1.71.0
Rome extension v0.14.1
```

### What happened?

1. Change a value in `rome.json` (e.g. switch from `"indentStyle": "tab"` to `"indentStyle": "space"` or vice-versa)
2. Go to a TS file and format.

The Rome extension uses the old config.

### Expected result

It would be really valuable if the extension automatically used the new `rome.json` contents immediately.

This sounds a bit like a "nice to have" feature, but it's critical for smooth UX when switching between commits that have different `rome.json` contents.

As a concrete example, I've just converted a project to Rome, and I've had to turn off about a dozen rules to avoid several hundred linter errors until I can tackle them one rule at a time: https://github.com/cubing/cubing.js/blob/4d4f6dbbdc4b4c69e88805edc74560ad3144abf0/rome.json#L10-L24

If I start a branch to turn the rules on one-by-one, then I need to be very careful to "Reload Window" in VSCode every time I switch to or from the branch. Otherwise, I will see false positives or false negatives in the editor based on a different `rome.json`. Even worse, any auto-formatting from the wrong `rome.json` can accidentally end up applying to another branch's committed changes, which could cause a massive merge conflict.

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

**Comment by lgarron** [comment]
If this is too tricky to implement a clean restart of the LSP, it would be sufficiently useful for me to have a way to using stale `rome.json` changes by accident, either through default behaviour or using any sort of configuration.

One option might be to detect `rome.json` changes and just *halt* the LSP, with a notification to [restart the Rome LSP](https://github.com/rome/tools/issues/3196) or "Reload Window" to resume Rome extension functionality.

**Comment by ematipico** [comment]
This is something that we plan to enable, but not very soon. In order to add this feature, the extension needs to add support for JSON files and doing so would automatically implement the formatting of that file. Rome is not able to support formatting for JSON files. If we decided to enable it now, an an error dialog would appear and... I am not sure it would send the correct message. When Rome will be able to support JSON files, this feature will be added.

I would categorize this as a feature, and the team plans to add it soon (2/3 months).

**Reply by leops** [reply]
I don't think we need JSON support for this, the Language Server Protocol supports registering arbitrary filesystem watchers from the server that would deliver a notification whenever the configuration file changes, we could use this to reload its content and update the workspace settings

**Reply by ematipico** [reply]
That's great, I didn't know it was possible!

**Comment by lgarron** [comment]
> I would categorize this as a feature, and the team plans to add it soon (2/3 months).

🥳

Definitely looking forward to something like that!
If this issue is closed, is there another issue/discussion I could follow to see when the change is being working on?

**Reply by ematipico** [reply]
I converted this issue into a discussion. We will create an issue once we have the technical details and capacity to implement the feature
